darwin/button.m

```diff
@@ -103,7 +103,7 @@ static void defaultOnClicked(uiButton *b, void *data)
 	uiDarwinSetControlFont(b->button, NSRegularControlSize);
 
 	if (buttonDelegate == nil) {
-		buttonDelegate = [buttonDelegateClass new];
+		buttonDelegate = [[buttonDelegateClass new] autorelease];
```

Comment:
In the above delegates code, clang's analyzer (Xcode > Product > Analyze) will warn:

> libui/darwin/alloc.m:31:3: Incorrect decrement of the reference count of an object that is not owned at this point by the caller

The array retains the objects, and thus becomes the owner. By releasing objects from the array directly, you are releasing an object that doesn't belong to you. You should really only release objects that you hold references to. Since we don't need the object after it's done being stored in the array, we can autorelease the object so that the array has the only reference. Since it doesn't get released until the app quits this is fine.

Autoreleasing should always be preferred over manual releases. One, it's not really more performant to release objects directly. You should use autorelease pools if you need performance (rare). Two, it's one step closer to Automatic Reference Counting, which ideally every Objective-C project should move to, as you'll get slight performance improvements, and better code.